api/tests/Behat/bootstrap/v2/Reporting/Sections/DebtsSectionTrait.php

```diff
@@ -0,0 +1,221 @@
+    private array $debtTypes = [
+        'care-fees' => [
+            'name' => 'Outstanding care home fees',
+            'amount' => '£0.00',
+        ],
+        'credit-cards' => [
+            'name' => 'Credit cards',
+            'amount' => '£0.00',
+        ],
+        'loans' => [
+            'name' => 'Loans',
+            'amount' => '£0.00',
+        ],
+        'other' => [
+            'name' => 'Other',
+            'amount' => '£0.00',
+        ],
+        'total-amount' => [
+            'name' => 'Total amount',
+            'amount' => '£0.00',
+        ],
+    ];
+
+    /**
+     * @When I view and start the debts report section
+     */
+    public function iViewAndStartDebtsSection()
+    {
+        $this->iVisitDebtsSection();
+        $this->pressButton('Start debts');
+    }
+
+    /**
+     * @When I have no debts
+     */
+    public function iHaveNoDebts()
+    {
+        $this->iAmOnDebtsExistPage();
+        $this->selectOption('yes_no[hasDebts]', 'no');
+        $this->pressButton('Save and continue');
+    }
+
+    /**
+     * @When I have a debt to add
+     */
+    public function iAddADebt()
+    {
+        $this->iAmOnDebtsExistPage();
+        $this->selectOption('yes_no[hasDebts]', 'yes');
+        $this->pressButton('Save and continue');
+    }
+
+    /**
+     * @When I add some debt values
+     */
+    public function iAddSomeDebtValues()
+    {
+        $this->hasDebts = true;
+        $this->addADebtsPayment('credit-cards', '1500');
+        $this->pressButton('Save and continue');
+    }
+
+    /**
+     * @When I say how the debts are being managed
+     */
+    public function iSayHowTheDebtsAreBeingManaged()
+    {
+        $this->iAmOnDebtsManagementPage();
+        $this->debtManagement[] = ['Lorem ipsum'];
+        $this->fillField('debtManagement[debtManagement]', 'Lorem ipsum');
+        $this->pressButton('Save and continue');
+    }
+
+    /**
+     * @Then I should see the expected debts section summary
+     */
+    public function iShouldSeeTheExpectedDebtsSummary()
+    {
+        $this->iAmOnDebtsSummaryPage();
+
+        if (!$this->hasDebts) {
+            $haveDebts[] = ['no'];
+        } else {
+            $haveDebts[] = ['yes'];
+        }
+
+        $this->expectedResultsDisplayed(0, $haveDebts, 'Answer for "Does user have any debts?"');
+
+        if ($this->hasDebts) {
+            // Convert the debts list amount into the correct format for comparison
+            $expectedDebtsList = [];
+            foreach ($this->debtTypes as $debtListKey => $debtList) {
+                /*
+                 * The total amount part of the debtTypes will have a small
+                 * difference when formatting the amount as the $this->totalAmount variable
+                 * is a float so we will have to cast it to a string before passing it
+                 * to the moneyFormat() function.
+                 */
+                foreach ($debtList as $debtItemKey => $debtItem) {
```

Comment:
Hopefully the comment above explains this `foreach` enough without having me personally explain it. If not I will look to change it